src/Cli/dotnet/Installer/Windows/InstallRequestMessage.cs

```diff
@@ -33,7 +33,7 @@ public string ManifestPath
         /// The contents of the install state file. Each element corresponds to a single line of
         /// the JSON file to be written.
         /// </summary>
-        public string[] InstallStateContents
+        public Dictionary<string, string> InstallStateContents
```

Comment:
Naming nit: I think `InstallStateManifestVersions` would be a bit clearer here.

```suggestion
        public Dictionary<string, string> InstallStateManifestVersions
```